web/src/components/CitySelector.tsx

```diff
@@ -52,19 +56,28 @@ const CitySelector = ({ cities, language, filterText, stickyTop = 0 }: PropsType
     const safeCities = cities.filter(city => city.code !== '')
     const sorted = sort(safeCities)
     const groups = groupBy(sorted, city => city.sortCategory)
-    return transform(
-      groups,
-      (result: Array<ReactNode>, cities, key) => {
-        result.push(
-          <div key={key}>
-            <CityListParent stickyTop={stickyTop}>{key}</CityListParent>
-            {cities.map(city => (
-              <CityEntry key={city.code} city={city} language={language} filterText={filterText} />
-            ))}
-          </div>
-        )
-      },
-      []
+    const failureProps = {
+      goToPath: null,
+      errorMessage: 'nothingFound'
+    }
+
+    return sorted.length === 0 ? (
+      <Failure {...failureProps} t={t} />
+    ) : (
```

Comment:
Optional: Imo it's more explicit and therefore better readable to use an regular if statement here